login_server/player/state/authenticated_state.py

```diff
@@ -352,10 +373,45 @@ def handle_client2login_connect(self, message: Client2LoginConnect):
         # The player is now known to be modded
         self.player.is_modded = True
         # Give the player their current mode
-        resp = Login2ClientModeInfo('ootb')
-        self._send_control_message(self.player, resp)
+        mode_info = Login2ClientModeInfo(self.player.player_settings.game_setting_mode)
+        self._send_control_message(self.player, mode_info)
+        # Give the player the appropriate class menu data
+        for data_point in get_stripped_class_menu_data(self.player.player_settings.game_setting_mode):
+            menu_data = Login2ClientMenuData(data_point)
+            self._send_control_message(self.player, menu_data)
+
+        for loadout_point in self.player.loadouts.strip_loadouts_for_modded_menus(self.player.player_settings
+                                                                                          .game_setting_mode):
+            loadout_data = Login2ClientLoadouts(loadout_point)
+            self._send_control_message(self.player, loadout_data)
 
     @handles_control_message(messageType=Client2LoginSwitchMode)
     def handle_client2login_switchmode(self, message: Client2LoginSwitchMode):
-        resp = Login2ClientModeInfo('ootb')
-        self._send_control_message(self.player, resp)
+        modes = list(get_game_setting_modes())
+        next_mode = modes[(modes.index(self.player.player_settings.game_setting_mode) + 1) % len(modes)]
+        self.player.player_settings.game_setting_mode = next_mode
+
+        # Send the player a message confirming their mode
+        self._send_private_msg_from_server(self.player, 'You are now in %s mode'
+                                           % self.player.player_settings.game_setting_mode)
+
+        # Send the control message indicating the switch
```

Comment:
Avoid duplication between this function and the previous one.

Reply:
Fixed